libminifi/src/core/state/nodes/FlowInformation.cpp

```diff
@@ -61,33 +67,74 @@ std::vector<SerializedResponseNode> FlowInformation::serialize() {
     serialized.push_back(queues);
   }
 
-  if (nullptr != monitor_) {
-    SerializedResponseNode componentsNode{.name = "components", .collapsible = false};
-    monitor_->executeOnAllComponents([&componentsNode](StateController& component){
-      componentsNode.children.push_back({
-        .name = component.getComponentName(),
+  if (!processors_.empty()) {
+    SerializedResponseNode processorsStatusesNode{.name = "processorStatuses", .array = true, .collapsible = false};
+    for (const auto processor : processors_) {
+      if (!processor) {
+        continue;
+      }
+
+      auto metrics = processor->getMetrics();
+      processorsStatusesNode.children.push_back({
+        .name = processor->getName(),
         .collapsible = false,
         .children = {
-          {.name = "running", .value = component.isRunning()},
-          {.name = "uuid", .value = std::string{component.getComponentUUID().to_string()}}
+          {.name = "id", .value = std::string{processor->getUUIDStr()}},
+          {.name = "groupId", .value = processor->getProcessGroupUUIDStr()},
+          {.name = "bytesRead", .value = metrics->bytes_read.load()},
+          {.name = "bytesWritten", .value = metrics->bytes_written.load()},
+          {.name = "flowFilesIn", .value = metrics->incoming_flow_files.load()},
+          {.name = "flowFilesOut", .value = metrics->transferred_flow_files.load()},
+          {.name = "bytesIn", .value = metrics->incoming_bytes.load()},
+          {.name = "bytesOut", .value = metrics->transferred_bytes.load()},
+          {.name = "invocations", .value = metrics->invocations.load()},
+          {.name = "processingNanos", .value = metrics->processing_nanos.load()},
+          {.name = "activeThreadCount", .value = -1},
+          {.name = "terminatedThreadCount", .value = -1},
+          {.name = "runningState", .value = (processor->isRunning() ? "RUNNING" : "STOPPED")}
```

Comment:
@martinzink can you confirm that this is compatible with what you proposed in nifi?

Reply:
I went with "runningState", but if you guys prefer "runState" or "runStatus" or anything else I'm okay with that too.

Reply:
I think its easiest if we follow the nifi api 
https://github.com/apache/nifi-api/blob/main/src/main/java/org/apache/nifi/controller/status/RunStatus.java
```suggestion
          {.name = "runStatus", .value = (processor->isRunning() ? "Running" : "Stopped")}
```

Reply:
Updated in https://github.com/apache/nifi-minifi-cpp/pull/1909/commits/3f4594ca8c113b1e60c04dc8f1c18e6c918ed920